src/tests/movie.test.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import Movies from '../modules/movies.js';
+
+global.fetch = jest.fn(() => Promise.resolve({
+  json: () => Promise.resolve(
+    [
+      {
+        show: {
+          id: 6490,
+          url: 'https://www.tvmaze.com/shows/6490/comedy-shuffle',
+          name: 'Comedy Shuffle',
+          type: 'Variety',
+          language: 'English',
+        },
+      },
+      {
+        show: {
+          id: 10428,
+          url: 'https://www.tvmaze.com/shows/10428/comedy-playhouse',
+          name: 'Comedy Playhouse',
+          type: 'Scripted',
+          language: 'English',
+        },
+      },
+    ],
+  ),
+}));
+
+describe('test get movies count method', () => {
+  const movies = new Movies();
+
+  test('test getMoviesCount I', async () => {
+    await movies.getMovieList('girls');
+    await movies.getMovieList('boys');
+
+    expect(movies.getMoviesCount()).toBe(4);
+  });
```

Comment:
- [ ]  I am afraid the test is not using the mock API call because the data passing through the promise and the test output is not matching.  I believe a better way to make a test on a function is to pass a parameter to the function we are testing. 

![image](https://user-images.githubusercontent.com/29539518/175213349-cb9dbba7-5c6c-4188-8f76-95acadb6b366.png)

As you can see here on the picture, there is a few seconds delay; I believe that's because it's fetching data from the API and not the mock.  This works, but as you know, data from API can fluctuate at any point, which might cause a fail on the test.

- please consider checking the function that calculates the number one more time.